### ISSUE TITLE
TP-LINK-login-Escalation-of-Privileges validity?

### ISSUE THREAD

**Issue by nnposter** [yough3rt/IOT-pwn-for-fun#1]
I might be missing something but are you sure that the alleged backdoor password truly works as such?

The way I interpreted the code the logic depicted in the advisory...
```lua
if not e then
  return true
end
if e==DEFAULT_PWD then
  return true
else
  return false
end
```
...in function `is_factory_passwd` is merely providing a flag that no real password has been set for the `admin` account. In other words, I cannot find anywhere a logic that says that access is allowed if the supplied password matches `DEFAULT_PWD`.

The real authentication function appears to be `action_login` in `dispatcher.lua`. It has the following logic:
1. If `is_factory_passwd()` returns `true` then the access is granted. Note that at this point the code does not even look at the submitted password. Any submitted password is implicitly accepted.
1. If the submitted user does not exist then the access is rejected.
1. If the submitted user account is locked then the access is rejected.
1. If the submitted password matches the password of the submitted user account then the access is granted.
1. The access is rejected.

In summary, the significance of user `admin` either not having a password set at all or having it set to `DEFAULT_PWD` appears to mean that authentication is effectively disabled. Submitting login combination `admin:zMiVw8Kw0oxKXL0` will not grant access if the account is set with a different password.

**Comment by yough3rt** [comment]
Thank you for first!
I reviewed the code deep again which indeed just as you said. It's my fault that just trying `admin:zMiVw8Kw0oxKXL0` while ignoring the root cause of it.
BTW, I donot think it is a good design to accept any submitted password if password is default.
However, I will remove the `TP-LINK-login-Escalation-of-Privileges` page, thank you anyway.
